homeassistant/bootstrap.py

```diff
@@ -172,12 +170,26 @@
     # add it here.
     "backup",
 }
-RECORDER_INTEGRATIONS = {
-    # Setup after frontend
-    # To record data
-    "recorder",
-}
-DISCOVERY_INTEGRATIONS = ("bluetooth", "dhcp", "ssdp", "usb", "zeroconf")
+# Stage 0 is divided into substages. Each substage has a name, a set of integrations and a timeout.
+STAGE_0_INTEGRATIONS = (
+    # Load logging and http deps as soon as possible
+    (
+        "logging, http deps",
+        LOGGING_AND_HTTP_DEPS_INTEGRATIONS,
+        STAGE_0_SUBSTAGE_TIMEOUT,
+    ),
+    # Setup frontend
+    ("frontend", FRONTEND_INTEGRATIONS, STAGE_0_SUBSTAGE_TIMEOUT),
+    # Setup recorder
+    ("recorder", {"recorder"}, STAGE_0_SUBSTAGE_TIMEOUT),
```

Comment:
How do we handle database migrations?

Reply:
The timers are frozen during database migration

Reply:
It seems it only freezes the timer for its own domain https://github.com/home-assistant/core/blob/954a57e5ac0d3e062cc68970c133dd8246b453e8/homeassistant/components/recorder/core.py#L932

Reply:
We should put a sleep longer than the in each of the important migrations and start up, to see what happens, to make sure the system can handle that type of failure